src/Dispatcher.php

```diff
@@ -136,10 +151,28 @@ public function process_scheduled_webhook( string $url, string $action, string $
 
 		// Check if the request was successful
 		if ( is_wp_error( $response ) || 200 !== wp_remote_retrieve_response_code( $response ) ) {
+			// Track the failure for internal monitoring and blocking logic
 			$this->handle_webhook_failure( $url, $response );
-		} else {
-			$this->handle_webhook_success( $url );
+
+			// Throw WP_Exception to signal failure to Action Scheduler
+			// This allows Action Scheduler to catch the exception and handle it appropriately,
+			// providing better failure tracking and retry logic within the scheduler itself.
+			$error_message = $this->get_failure_error_message( $response );
+			throw new WP_Exception(
+				'webhook_delivery_failed',
+				sprintf(
+					/* translators: 1: webhook URL, 2: action, 3: entity, 4: ID, 5: error message */
+					__( 'Webhook delivery failed for URL %1$s (action: %2$s, entity: %3$s, ID: %4$s): %5$s', 'wp-webhook-framework' ),
+					$url,
+					$action,
+					$entity,
+					$id,
+					$error_message
+				)
```

Comment:
This is still not only one message. WP_Exception extends the normal exception class and therefore accepts one parameter for the message. No error code. 

https://developer.wordpress.org/reference/classes/wp_exception/

public [__construct](https://www.php.net/manual/en/exception.construct.php)([string](https://www.php.net/manual/en/language.types.string.php) $message = "", [int](https://www.php.net/manual/en/language.types.integer.php) $code = 0, [?](https://www.php.net/manual/en/language.types.null.php)[Throwable](https://www.php.net/manual/en/class.throwable.php) $previous = [null](https://www.php.net/manual/en/reserved.constants.php#constant.null))